aptos-move/aptos-aggregator/src/resolver.rs

```diff
@@ -18,29 +28,89 @@ pub enum AggregatorReadMode {
 /// Returns a value of an aggregator from cache or global storage.
 ///   - Ok(..)       if aggregator value exists
 ///   - Err(..)      otherwise.
-pub trait AggregatorResolver {
+pub trait TAggregatorView {
+    type Identifier;
+
+    fn get_aggregator_v1_state_value(
+        &self,
+        id: &Self::Identifier,
+    ) -> anyhow::Result<Option<StateValue>>;
+
+    fn get_aggregator_v1_value(&self, id: &Self::Identifier) -> anyhow::Result<u128> {
```

Comment:
We don't need read mode for aggregator v1?
Should we set speculative-start-value to 0 if the aggregator is v1, and set speculative-start-value to cheap read if the aggregator is v2?

Reply:
I also think that we didn't want to change the reading mode for V1, but let's double check.

Reply:
aaaactually :) 

I think we have/want to have same native code logic in the extension for both V1 and V2. And so V1 aggregator also does a cheap read, and does branch prediction, etc. 
We do throw an exception if add/sub fail in V1 (even though that should never happen)

It only gets separated once VMChangeSet gets computed. 

Is that OK?

Reply:
we could have different logic inside the read based on the ID, though.

we cannot start with speculative_start_value = 0, in case we are doing more sub than add.

we could most likely start with speculative_start_value = MAX_U64 / 2, but that seems risky.
I think it is better to just have cheap read for V1 as well

Reply:
It can still fail speculatively w. sub or add (even with an aggregated read), just at that point for V1 we know we can fail the txn because the read it based on was wrong - in the correct/committing state it can never over/underflow. So it should be Ok to share the speculative read logic in there, have aggregatorV1 native make the call to try_add try_sub and abort the txn if it that call doesn't succeed.
@georgemitenkov can you add back the mode parameter for V1 then?